podtailer/podtailer.go

```diff
@@ -77,12 +80,27 @@ loop:
 				logrus.WithError(err).Error("Error setting up watcher")
 				continue loop
 			}
+			logrus.WithFields(logrus.Fields{
+				"name":      pod.Name,
+				"uid":       pod.UID,
+				"namespace": pod.Namespace,
+			}).Info("starting watcher for pod")
 			watcher.Start()
-			defer watcher.Stop()
+			watcherMap[pod.UID] = watcher
+		case deletedPodUID := <-podWatcher.DeletedPods():
+			logrus.WithFields(logrus.Fields{
+				"uid": deletedPodUID,
+			}).Info("pod deleted, stopping watcher")
+			watcherMap[deletedPodUID].Stop()
```

Comment:
Might want to first check that `deletedPodUID` is present in the `watcherMap` here, to avoid a possible nil pointer dereference. (Not sure if we can actually get a delete for a pod we're not already watching, but might as well be defensive)

Reply:
Oops, good catch! Fixed